apps/service-portal/src/components/UserDelegations/UserDelegations.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useMemo } from 'react'
+import { getUserManager, useAuth, User } from '@island.is/auth/react'
+import { Stack, Text, TopicCard } from '@island.is/island-ui/core'
+import { useLocale } from '@island.is/localization'
+import {
+  ActorDelegationsQuery,
+  useActorDelegationsQuery,
+} from '@island.is/service-portal/graphql'
+
+// The actor name and nationalId should be added to userinfo endpoint so we
+// can get rid of this mess.
+interface JoseUtil {
+  parseJwt(
+    token: string,
+  ): {
+    payload: {
+      nationalId: string
+      act?: {
+        nationalId: string
+      }
+    }
+  }
+}
+const getActorNationalId = (userInfo: User | null) => {
+  if (!userInfo) {
+    return null
+  }
+  const jose = (getUserManager() as any)._joseUtil as JoseUtil
+  const jwt = jose.parseJwt(userInfo.access_token)
+  return jwt.payload.act?.nationalId ?? jwt.payload.nationalId
+}
+
+export const UserDelegations = () => {
+  const { formatMessage } = useLocale()
+  const { userInfo, switchUser } = useAuth()
+  const { data } = useActorDelegationsQuery()
+  const currentNationalId = userInfo?.profile.nationalId as string
+  const actorNationalId = useMemo(() => getActorNationalId(userInfo), [
+    userInfo,
+  ])
+  const hasData = data && data.authActorDelegations.length > 0
+  const isDelegation = currentNationalId !== actorNationalId
+
+  if (!hasData && !isDelegation) {
+    return null
+  }
+
+  let delegations = [] as ActorDelegationsQuery['authActorDelegations']
+  if (data) {
+    delegations.push(...data.authActorDelegations)
+  }
+  if (isDelegation) {
+    delegations = delegations.filter(
+      (delegation) => delegation.fromNationalId !== currentNationalId,
+    )
+    delegations.unshift({ fromName: 'Ég', fromNationalId: actorNationalId })
```

Comment:
Don't we want translations for `Ég`?

Reply:
Ah, forgot to explain this.

This is temporary until we get the name of the real user in the id token or userinfo endpoint.